third-party/realdds/include/realdds/dds-stream-profile.h

```diff
@@ -83,39 +48,52 @@ class dds_stream_base;
 
 class dds_stream_profile
 {
-    dds_stream_uid _uid;
-    dds_stream_format _format;
     int16_t _frequency;  // "Frames" per second
+    dds_stream_format _format;
 
     std::weak_ptr< dds_stream_base > _stream;
 
 public:
     virtual ~dds_stream_profile() {}
 
 protected:
-    dds_stream_profile( dds_stream_uid uid, dds_stream_format format, int16_t frequency )
-        : _uid( uid )
-        , _format( format )
+    dds_stream_profile( dds_stream_format format, int16_t frequency )
+        : _format( format )
         , _frequency( frequency )
     {
     }
     dds_stream_profile( dds_stream_profile && ) = default;
+    dds_stream_profile( nlohmann::json const &, int & index );
 
 public:
     std::shared_ptr< dds_stream_base > stream() const { return _stream.lock(); }
     // This is for initialization and is called from dds_stream_base only!
     void init_stream( std::weak_ptr< dds_stream_base > const & stream );
 
-    dds_stream_uid uid() const { return _uid; }
     dds_stream_format format() const { return _format; }
     int16_t frequency() const { return _frequency; }
 
     // These are for debugging - not functional
     virtual std::string to_string() const;
     virtual std::string details_to_string() const;
 
-    // Serialization to a JSON representation
+    // Serialization to a JSON array representation
     virtual nlohmann::json to_json() const;
+
+    // Build a profile from a json array object, e.g.:
+    //      auto profile = dds_stream_profile::from_json< dds_video_stream_profile >( j );
+    // This is the reverse of to_json() which returns a json array
+    template< class final_stream_profile >
+    static std::shared_ptr< final_stream_profile > from_json( nlohmann::json const & j )
+    {
+        int it = 0;
+        auto profile = std::make_shared< final_stream_profile >( j, it );
+        verify_end_of_json( j, it );  // just so it's not in the header

```

Comment:
Why not put all of `from_json` implementation in the cpp? It is not inlined anyway